arangod/Aql/ClusterNodes.cpp

```diff
@@ -415,15 +415,11 @@ SingleRemoteOperationNode::SingleRemoteOperationNode(ExecutionPlan* plan,
   } else if (_mode == NodeType::REMOVE) {
     TRI_ASSERT(!_key.empty());
     TRI_ASSERT(_inVariable == nullptr);
-    TRI_ASSERT(_outVariable == nullptr);
     TRI_ASSERT(_outVariableNew == nullptr);
   } else if (_mode == NodeType::INSERT) {
     TRI_ASSERT(_key.empty());
-    TRI_ASSERT(_outVariable == nullptr);
   } else if (_mode == NodeType::UPDATE) {
-    TRI_ASSERT(_outVariable == nullptr);
   } else if (_mode == NodeType::REPLACE) {
-    TRI_ASSERT(_outVariable == nullptr);
```

Comment:
What's the reason for removing these assertions? It's not immediately clear to me why this should change. In any case, if we remove them, we should remove the empty `else if` clauses as well.

Reply:
Now we set the `_outVariable` so we can not assert it to be `nullptr` anymore. I have left the `else if` so we do not end up in the `else` case that `asserts false`.

Reply:
Okay, I see how we're setting it now. Can we clean up the `if` tree though? Instead of:
```
} else if (_mode == NodeType::UPDATE) {
} else if (_mode == NodeType::REPLACE) {
} else {
```
it could just be 
```
} else if (_mode != NodeType::UPDATE && _mode != NodeType::REPLACE) {
```

Reply:
I did this in a separate PR:
https://github.com/arangodb/arangodb/pull/7658